src/game/server/props.cpp

```diff
@@ -1675,22 +1680,19 @@ void CBreakableProp::Break( CBaseEntity *pBreaker, const CTakeDamageInfo &info )
 	bool bExploded = false;
 
 	CBaseEntity *pAttacker = info.GetAttacker();
-	if ( m_hLastAttacker )
-	{
-		// Pass along the person who made this explosive breakable explode.
-		// This way the player allies can get immunity from barrels exploded by the player.
-		pAttacker = m_hLastAttacker;
-	}
-	else if( m_hPhysicsAttacker )
+	if ( HasPhysicsAttacker() )
```

Comment:
Why are we overriding m_hLastAttacker with `HasPhysicsAttacker()`? If `Break` is called from `Event_Kill` then this will override the person who actually caused the damage with the person holding the prop with the physgun. If someone is holding an exploding barrel, wouldn't this make them the attacker in this case?